feat: Spinner 컴포넌트 완료

KimChunsick opened the pull request:
feat: Spinner 컴포넌트 작업 완료 및 blog에서 확인 가능하게 컴포넌트 추가

Signed-off-by: deokwon <deokwon@deokwonui-MacBookPro.local>

Commit:
feat: Spinner 컴포넌트 완료

feat: Spinner 컴포넌트 작업 완료 및 blog에서 확인 가능하게 컴포넌트 추가

Signed-off-by: deokwon <deokwon@deokwonui-MacBookPro.local>

Vallista commented on packages/core/src/components/Spinner/Spinner.tsx:
```diff
@@ -0,0 +1,72 @@
+
+  return (
+    <SpinnerWrapper size={size}>
+      <div>
```
div가 왜필요함요? SpinnerWrapper가 div라 `<></>` Fragment로 충분히 가능한데요

Vallista replied:
그리고 딱히 없어도 display flex를 이용해서 중앙정렬 박고 하면 될 것 같은데 이렇게 한 이유가 무엇입니까요?

KimChunsick replied:
vercel spinner 컴포넌트의 html, css를 고대로 가져와서 그렇습니다.
수정하겠습니다.

Vallista commented on packages/core/src/components/Spinner/Spinner.tsx:
```diff
@@ -0,0 +1,72 @@
+    <SpinnerWrapper size={size}>
+      <div>
+        {[...Array(12)].map((_, index) => {
+          return <SpinnerStick key={index} index={index} />
```
key에 index를 주는건 리액트에서의 안티패턴입니다.

https://robinpokorny.medium.com/index-as-a-key-is-an-anti-pattern-e0349aece318

애초에 `리스트 항목에 명시적으로 key를 지정하지 않으면 React는 기본적으로 인덱스를 key로 사용합니다.` 임. ([공식 문서 참조](https://ko.reactjs.org/docs/lists-and-keys.html#keys))

KimChunsick replied:
확인했습니다. 감사합니다.

Vallista commented on packages/core/src/components/Spinner/type.ts:
```diff
@@ -0,0 +1,11 @@
+  size: number
+}
+
+export type ReturningUseSpinner<T extends Partial<SpinnerProps> = Partial<SpinnerProps>> = T & {
```
이건 Text 처럼 쓴 이유가 뭥미?

굳이 필요없을 것 같은데..

KimChunsick replied:
Text의 구조를 그대로 반영하는게 좋지 않을까 해서 사용해봤습니다.

Vallista commented on packages/core/src/components/Spinner/useSpinner.ts:
```diff
@@ -0,0 +1,11 @@
+import { Colors } from '../ThemeProvider'
+import { ReturningUseSpinner, SpinnerProps } from './type'
+
+const initProps: Partial<SpinnerProps> = {}
```
이 친구 안쓰면 필요가 있나요..?

차라리 저위에 width, height를 여기서 default로 20씩 지정해줄거면 ㅇㅈ

KimChunsick replied:
수정하겠습니다.

Vallista reviewed (changes requested):
확인하십시오.

Commit:
fix: 리뷰 수정 내용 반영

fix: SpinnerStick의 key 값을 dom 전역에서 unique 하게끔 수정
fix: SpinnerWrapper와 SpinnerStick 사이의 필요없는 div element 삭제
fix: initProps에 size prop의 기본값을 추가

Signed-off-by: deokwon <deokwon@deokwonui-MacBookPro.local>

Vallista reviewed (approved):
허가한다.
